tests/integration/test_fim/test_synchronization/test_sync_time.py

```diff
@@ -0,0 +1,162 @@
+    '''
+    description: Check when the 'wazuh-syscheckd' daemon is performing a synchronization, a normal synchronization
+                 will end before the configured `interval` and `max_interval`.
+
+    test_phases:
+        - Create a folder with a number of files inside.
+        - Restart syscheckd.
+        - Check that a sync interval started, and get the time it starts
+        - Get all the integrity state events time.
+        - Assert that the time it took for the sync to complete was less than the configured interval and max_interval.
+
+    wazuh_min_version: 4.5.0
+
+    tier: 2
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration values for ossec.conf.
+        - metadata:
+            type: dict
+            brief: Test case data.
+        - set_wazuh_configuration_fim:
+            type: fixture
+            brief: Set ossec.conf and local_internal_options configuration.
+        - create_files_in_folder:
+            type: fixture
+            brief: create files in monitored folder, and delete them after the test.
+        - restart_syscheck_function:
+            type: fixture
+            brief: restart syscheckd daemon, and truncate the ossec.log.
+        - wait_for_fim_start_function:
+            type: fixture
+            brief: check that the starting fim scan is detected.
+
+    assertions:
+        - Assert sync time delta is smaller than interval
+        - Assert sync time delta is smaller than max_interval
+
+    input_description: A test case is contained in external YAML file (cases_sync_interval.yaml) which includes
+                       configuration settings for the 'wazuh-syscheckd' daemon. That is combined with the interval
+                       periods and the testing directory to be monitored defined in this module.
+
+    expected_output:
+        - r'Initializing FIM Integrity Synchronization check'
+        - r".*Executing FIM sync"
+        - r".*Sending integrity control message.*"
+
+    tags:
+        - scheduled
+    '''
+
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Wait for new sync and get start time
+    sync_time = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                        callback=evm.callback_sync_start_time,
+                                        error_message=fim.ERR_MSG_FIM_SYNC_NOT_DETECTED, update_position=True).result()
+
+    # Get the time of all the sync state events for the created files
+    results = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                      callback=evm.callback_state_event_time, accum_results=3,
+                                      error_message=fim.ERR_MSG_FIM_SYNC_NOT_DETECTED, update_position=True).result()
+
+    # Calculate timedelta between start of sync and last message.
+    delta = (results[-1] - sync_time).total_seconds() + 1
```

Comment:
We need to validate that it not generate issues.

Reply:
Added comment in https://github.com/wazuh/wazuh-qa/pull/3380/commits/6b66bc842d17c0bc1f7d20dc3dcc8c96bce3d18e